process.c

```diff
@@ -8534,6 +8535,37 @@ static VALUE rb_mProcUID;
 static VALUE rb_mProcGID;
 static VALUE rb_mProcID_Syscall;
 
+/*
+ *  call-seq:
+ *     Process.warmup    -> true
+ *
+ *  Notify the Ruby virtual machine that the boot sequence is finished,
+ *  and that now is a good time to optimize the application. This is useful
+ *  for long running applications.
+ *
+ *  This method is expected to be called at the end of the application boot.
+ *  If the application is deployed using a pre-forking model, +Process.warmup+
+ *  should be called in the original process before the first fork.
+ *
+ *  The actual optimizations performed are entirely implementation specific
+ *  and may change in the future without notice.
+ *
+ *  On CRuby, +Process.warmup+:
+ *
+ *  * Perform a major GC.
+ *  * Compacts the heap.
```

Comment:
Can we add a test for this two? I think we just need to check that `GC.stat(:major_gc_count)` and `GC.stat(:compact_count)` both increased by 1.